spacy/cli/convert.py

```diff
@@ -110,6 +112,14 @@ def convert(
     input_path = Path(input_path)
     if not msg:
         msg = Printer(no_print=silent)
+
+    # Add warnings for renamed language code in v4
+    renamed_lang_codes = {"xx": "mul", "is": "isl"}
+    if lang in renamed_lang_codes:
+        warnings.warn(
```

Comment:
In the CLI commands, use `msg.warn` instead of `warnings.warn` and `msg.fail` instead of raising errors.

I think that a warning isn't enough here (since you're not automatically remapping) and it should fail instead? I think we'd want `msg.fail(exits=1)` so that people switch to the updated codes and aren't surprised to have different codes in their codes than they specified. (I could potentially be convinced otherwise, but I think we don't want to deal with the confusion once you get to overrides like `--nlp.lang xx` and it's cleaner to do it right at the beginning.)

We usually use hard-coded messages in the CLI commands instead of `Errors`/`Warnings`, but repeating this in so many places also seems like a bad idea.

I think the language code map could go into `spacy/errors.py` just like `OLD_MODEL_SHORTCUTS`.

I think `lang` is used by: `convert`, `init config`, and `init vectors`

Reply:
I created `RENAMED_LANGUAGE_CODES` in `errors.py` and replaced all errors/warnings with `msg.fail(exits=1)` for `convert`, `init config`, and `init vectors`. The tracebacks should be much more clearer now, we could maybe adjust the error text message a bit.